reciperadar/admin/products.py

```diff
@@ -73,8 +82,11 @@ def get_list(
         return len(results), results
 
     def on_model_change(self, form, model, is_created):
+        for name in model.names:
+            if name.id is None:
+                name.id = ProductName.generate_id()
         if is_created:
-            model.id = model.singular.replace(" ", "_").replace("-", "_")
+            model.id = model.singular_names[0].replace(" ", "_").replace("-", "_")
```

Comment:
We should use this as an opportunity to begin using opaque generated IDs at the product-level here.